enterprise/frontend/src/metabase-enterprise/clean_up/CleanupCollectionModal/CleanupCollectionModal.tsx

```diff
@@ -0,0 +1,178 @@
+import { usePagination } from "./hooks";
+import {
+  type DateFilter,
+  getDateFilterLabel,
+  getDateFilterValue,
+  itemKeyFn,
+} from "./utils";
+
+interface CleanupCollectionModalProps {
+  onClose: () => void;
+  params: { slug: string };
+}
+
+export const CleanupCollectionModal = ({
+  onClose: handleClose,
+  params: { slug },
+}: CleanupCollectionModalProps) => {
+  const dispatch = useDispatch();
+  const collectionId = Urls.extractCollectionId(slug);
+
+  // selection
+  const selection = useListSelect(itemKeyFn);
+
+  // pagination
+  const pagination = usePagination({ initialPage: 0, pageSize: 10 });
+
+  // sorting
+  const [sortOptions, setSortOptions] = useState<SortingOptions>({
+    sort_column: "name",
+    sort_direction: SortDirection.Asc,
+  });
+
+  const handleSortingChange = (sortingOpts: SortingOptions) => {
+    setSortOptions(sortingOpts);
+    pagination.resetPage();
+  };
+
+  // filters
+  const [dateFilter, setDateFilter] = useState<DateFilter>("tomorrow");
+  const handleChangeDateFilter = (nextDateFilter: DateFilter) => {
+    setDateFilter(nextDateFilter);
+    pagination.resetPage();
+    selection.clear();
+  };
+  const before_date = useMemo(() => {
+    return getDateFilterValue(dateFilter);
+  }, [dateFilter]);
+
+  const [recursiveFilter, setRecursiveFilter] = useState(false);
+  const handleChangeRecursiveFilter = (recursiveFilter: boolean) => {
+    setRecursiveFilter(recursiveFilter);
+    pagination.resetPage();
+    selection.clear();
+  };
+
+  // data
+  const {
+    data: staleItemsData,
+    isLoading,
+    error,
+  } = useListStaleCollectionItemsQuery(
+    collectionId
+      ? {
+          id: collectionId,
+          is_recursive: recursiveFilter,
+          before_date,
+          ...pagination.paginationFilters,
+          ...sortOptions,
+        }
+      : skipToken,
+  );
+
+  const itemsData = staleItemsData?.data;
+  const total = staleItemsData?.total ?? 0;
+  const items: StaleCollectionItem[] = useMemo(() => {
+    return (itemsData ?? []).map(item => Search.wrapEntity(item, dispatch));
+  }, [itemsData, dispatch]);
+
+  // selection cont.
+  const { getIsSelected } = selection;
+  const hasUnselected = useMemo(() => {
+    return items.some(item => !getIsSelected(item));
+  }, [getIsSelected, items]);
+
+  // pagination cont.
+  const { setTotal } = pagination;
+  useEffect(() => {
+    setTotal(total);
+  }, [setTotal, total]);
+
+  return (
+    <Modal.Root
+      opened
+      onClose={handleClose}
+      data-testid="cleanup-collection-modal"
+      size={1118}
+    >
+      <Modal.Overlay />
+      <Modal.Content
+        className={CS.modalContent}
+        mih={isLoading ? "25rem" : undefined}
+      >
+        <Modal.Header
+          px="2.5rem"
+          pt="2rem"
+          pb="1.5rem"
+          className={CS.modalHeader}
+        >
+          <Modal.Title fz="20px">{t`Select unused items to trash`}</Modal.Title>
```

Comment:
```suggestion
          <Modal.Title fz="20px">{t`Select unused items to move to trash`}</Modal.Title>
```

At least to me, "trash" as a verb isn't super clear

Reply:
Discussion of "to trash": https://metaboat.slack.com/archives/C01MS7DQKR6/p1710874586643169

Reply:
Wow, really wished I had seen that thread before. Great feedback and changed!

Reply:
"Trash" or "trash"?